synapse/rest/client/v1/push_rule.py

```diff
@@ -163,7 +164,7 @@ def notify_user(self, user_id):
         stream_id, _ = self.store.get_push_rules_stream_token()
         self.notifier.on_new_event("push_rules_key", stream_id, users=[user_id])
 
-    def set_rule_attr(self, user_id, spec, val):
+    def set_rule_attr(self, user_id, spec, val) -> Awaitable:
```

Comment:
Can we make this `async def` and await inside? I think that helps with stack traces

Reply:
Sure! We use this pattern quite a lot, unfortunately. Not sure if we will always want to change it or not. (It also makes the typing MUCH clearer IMO.)

Reply:
Yeah, i think we almost always do. I think having `Awaitable` in the type signature is probably something we should view as code smell

Reply:
I think you just made a future PR of mine much larger. 😉